h/static/scripts/directive/annotation.js

```diff
@@ -111,13 +109,21 @@ function updateDomainModel(domainModel, vm, permissions) {
 }
 
 /** Update the view model from the domain model changes. */
-function updateViewModel($scope, time, domainModel, vm, permissions) {
+function updateViewModel($scope, time, domainModel,
+                         vm, permissions, features) {
 
   vm.form = {
     text: domainModel.text,
     tags: viewModelTagsFromDomainModelTags(domainModel.tags),
   };
-  vm.annotationURI = new URL('/a/' + domainModel.id, vm.serviceUrl).href;
+
+  if (features.flagEnabled('direct_linking') &&
```

Comment:
You don't even need to check the feature flag here. We can simply include or omit the `incontext` link server-side when we toggle the flag, so would suggest making `domainModel.links && domainModel.links.incontext` your test.